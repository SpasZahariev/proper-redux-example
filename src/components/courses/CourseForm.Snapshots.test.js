import React from "react";
import renderer from "react-test-renderer";
import { authors, courses } from "../../../tools/mockData";
import CourseForm from "./CourseForm";

it("sets submit button label to 'Saving...' when saving prop is true", () => {
  const tree = renderer.create(
    <CourseForm
      course={courses[0]}
      authors={authors}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving={true}
    ></CourseForm>
  );

  //snapshots are the output of the component rendering
  //they are stored in the _snapshots_ folder when the test is run
  expect(true).toMatchSnapshot();
});

it("sets submit button label to 'Save' when saving prop is false", () => {
  const tree = renderer.create(
    <CourseForm
      course={courses[0]}
      authors={authors}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving={false}
    ></CourseForm>
  );

  expect(true).toMatchSnapshot();
});
