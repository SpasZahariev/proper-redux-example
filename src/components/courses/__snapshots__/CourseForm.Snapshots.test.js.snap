// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets submit button label to 'Save' when saving prop is false 1`] = `true`;

exports[`sets submit button label to 'Saving...' when saving prop is true 1`] = `true`;
